ScoutSuite/providers/aws/facade/utils.py

```diff
@@ -8,25 +8,34 @@ class AWSFacadeUtils:
     _clients = {}
 
     @staticmethod
-    async def get_all_pages(service: str, region: str, session: boto3.session.Session, paginator_name: str, response_key: str, **paginator_args):
-        client = AWSFacadeUtils.get_client(service, region, session)
+    async def get_all_pages(service: str, region: str, session: boto3.session.Session, paginator_name: str, entity: str, **paginator_args):
+        results = await AWSFacadeUtils.get_multiple_entities_from_all_pages(service, region, session, paginator_name, [entity], **paginator_args)
+        return results[entity]
+
+    @staticmethod
+    async def get_multiple_entities_from_all_pages(service: str, region: str, session: boto3.session.Session, paginator_name: str, entities: list, **paginator_args):
+        client = AWSFacadeUtils.get_client(service, session, region)
+
         # Building a paginator doesn't require any API call so no need to do it concurrently:
-        paginator = client.get_paginator(paginator_name).paginate(**paginator_args)
+        paginator = client.get_paginator(
+            paginator_name).paginate(**paginator_args)
 
         # Getting all pages from a paginator requires API calls so we need to do it concurrently:
-        return await run_concurrently(lambda: AWSFacadeUtils._get_all_pages_from_paginator(paginator, response_key))
+        return await run_concurrently(lambda: AWSFacadeUtils._get_all_pages_from_paginator(paginator, entities))
 
     @staticmethod
-    def _get_all_pages_from_paginator(paginator, key):
-        resources = []
+    def _get_all_pages_from_paginator(paginator, entities: list):
+        resources = {entity: [] for entity in entities}
+
         # There's an API call hidden behind each iteration:
         for page in paginator:
-            resources.extend(page[key])
+            for entity in entities:
+                resources[entity].extend(page[entity])
 
         return resources
 
     @staticmethod
-    def get_client(service: str, region: str, session: boto3.session.Session):
+    def get_client(service: str, session: boto3.session.Session, region: str = None):
```

Comment:
Only leaving picky insignificant comment because I want you to know someone read your ~50 commits PR :heart: There shouldn't be a space in front or after the equals sign I believe : `str=None`

Reply:
WOW thanks, means a lot 💓 . 
```suggestion
    def get_client(service: str, session: boto3.session.Session, region: str=None):
```